controllers/dspipeline_params.go

```diff
@@ -0,0 +1,210 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+	mf "github.com/manifestival/manifestival"
+	dspipelinesiov1alpha1 "github.com/opendatahub-io/ds-pipelines-controller/api/v1alpha1"
+	"math/rand"
+	"time"
+)
+
+const (
+	defaultDBHostPrefix               = "mariadb"
+	defaultDBHostPort                 = "3306"
+	defaultMinioHostPrefix            = "minio"
+	defaultMinioPort                  = "9000"
+	defaultArtifactScriptConfigMap    = "ds-pipeline-artifact-script-sample"
+	defaultArtifactScriptConfigMapKey = "artifact_script"
+	defaultDSPServicePrefix           = "ds-pipeline"
+)
+
+type DSPipelineParams struct {
+	Name                             string
+	Namespace                        string
+	Owner                            mf.Owner
+	ApiServerImage                   string
+	PipelineRuntime                  string
+	StripEOF                         string
+	ArtifactScript                   string
+	ArtifactImage                    string
+	InjectDefaultScript              string
+	ApplyTektonCustomResource        string
+	TerminateStatus                  string
+	AutoUpdatePipelineDefaultVersion string
+	DBConfigCONMAXLifetimeSec        string
+	VisualizationServerServiceHost   string
+	VisualizaionServerServicePort    string
+	ObjectStoreConfigBucketName      string
+	S3CredentialsSecretName          string
+	AccessKeySecretKey               string
+	SecretKey                        string
+	ObjectStoreConfigSecure          string
+	MinioServiceServiceHost          string
+	MinioServiceServicePort          string
+	ArtifactBucket                   string
+	ArtifactEndpoint                 string
+	ArtifactEndpointScheme           string
+	ArtifactLogs                     string
+	ArtifactTracking                 string
+	ArtifactScriptConfigMapName      string
+	ArtifactScriptConfigMapKey       string
+	DBUser                           string
+	DBPasswordSecretKey              string
+	DBPasswordSecret                 string
+	DBName                           string
+	DBHost                           string
+	DBPort                           string
+	DBPassword                       string
+	SecretKeyVal                     string
+	AccessKey                        string
+	MinioImage                       string
+	DBServiceName                    string
+	MinioServiceName                 string
+	PersistenceAgentImage            string
+	ApiServerServiceName             string
+	ScheduledWorkflowImage           string
+	VisualizationServerImage         string
+	ViewerCrdImage                   string
+	CronScheduleTimezone             string
+	MLPipelineUIImage                string
+	MaxNumViewer                     string
+}
+
+func passwordGen(n int) string {
+	rand.Seed(time.Now().UnixNano())
+	var chars = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890")
```

Comment:
Why not add in some symbols here too? :-) 
`~=+%^*/()[]{}/!@#$?|`

Reply:
I was hesitant to include any special characters since I wasn't sure if minio has any restrictions on this front, I think this should be a github issue and not scope of this pr if we want a more complex password.